### PR TITLE
add createScalarFunction to create a user-defined scalar function

### PR DESCRIPTION
**Pull request by vrischmann**
This is the first part of adding support for [user-defined functions](https://www.sqlite.org/c3ref/create_function.html).
This PR only implements scalar functions, aggregate functions will come later.

Partly addresses #27

### CODE REVIEW

**Comment by vrischmann** [comment]
It only _partly_ fixes #27 because I didn't implement saving/loading struct as blobs, I'm not sure if that's something that will always work ?

@daurnimator I know it's been a long time but if you're still interested in this feature I'd be interested in your feedback.

File: sqlite.zig [hunk 1]
```diff
@@ -590,6 +590,141 @@ pub const Db = struct {
...
+    ///
+    /// The return type of the function is converted to a SQLite value according to the same rules but reversed.
+    ///
+    pub fn createScalarFunction(self: *Self, func_name: [:0]const u8, comptime func: anytype, comptime create_flags: CreateFunctionFlag) !void {
+        const Type = @TypeOf(func);
+
+        const type_info = @typeInfo(Type);
+        if (type_info != .Fn) @compileError("expecting a function");
+
+        const fn_info = type_info.Fn;
```
Comment (daurnimator) [inline sqlite.zig:618]:
```suggestion
        const fn_info = switch(@typeInfo(Type)) {
            .Fn => |fn| fn,
            else => @compileError("expecting a function"),
        };
```

File: sqlite.zig [hunk 2]
```diff
@@ -590,6 +590,150 @@ pub const Db = struct {
...
+                }
+
+                fn bindValue(comptime ArgType: type, arg: *ArgType, sqlite_value: *c.sqlite3_value) void {
+                    switch (ArgType) {
+                        Text => arg.*.data = sliceFromValue(sqlite_value),
+                        Blob => arg.*.data = sliceFromValue(sqlite_value),
+                        else => switch (@typeInfo(ArgType)) {
+                            .Int => |info| if (info.bits <= 32) {
+                                const value = c.sqlite3_value_int(sqlite_value);
+                                arg.* = @intCast(ArgType, value);
```
Comment (daurnimator) [inline sqlite.zig:655]: Need to watch out here for signed vs unsigned.

File: sqlite.zig [hunk 3]
```diff
@@ -590,6 +590,150 @@ pub const Db = struct {
...
+            func_name,
+            fn_info.args.len,
+            flags,
+            null,
+            struct {
+                fn sliceFromValue(sqlite_value: *c.sqlite3_value) []const u8 {
+                    const size = @intCast(usize, c.sqlite3_value_bytes(sqlite_value));
+
+                    const value = c.sqlite3_value_text(sqlite_value);
+                    debug.assert(value != null); // TODO(vincent): how do we handle this properly ?
```
Comment (daurnimator) [inline sqlite.zig:643]: This function (and `bindValue`; and hence also `xFunc`) should probably return an error set that includes e.g. `OutOfMemory`

Comment (vrischmann) [reply sqlite.zig:653]: I don't think I can do that, `xFunc` can't return anything

**Pull request merged**